lib/Driver/ToolChains.cpp

```diff
@@ -1213,51 +1254,37 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
 
   // Select the linker to use
   StringRef Linker;
```

Comment:
StringRef is a type that refers to storage owned by someone else. That's fine for constant strings, but the new `getDefaultLinker` API you're setting up allows returning arbitrary strings. Can you change this to `std::string` instead?

(If this were a performance bottleneck I'd ask you to have the method write into a SmallString output parameter instead, but I doubt it'll matter here.)